src/librustc_mir/hair/pattern/check_match.rs

```diff
@@ -225,7 +225,15 @@ impl<'a, 'tcx> MatchVisitor<'a, 'tcx> {
                 let scrutinee_is_uninhabited = if self.tcx.features().exhaustive_patterns {
                     self.tcx.is_ty_uninhabited_from(module, pat_ty)
                 } else {
-                    self.conservative_is_uninhabited(pat_ty)
+                    if self.tcx.features().better_divergence_checking {
```

Comment:
Maybe we should lump this feature gate in with `exhaustive_patterns`? I'm not sure yet that I see the difference here

Reply:
Well, the one reason it might be worth separating is because the change to uninhabited function parameter warning may not really be well suited to `exhaustive_patterns`. But it does seem like overkill to have another feature for this, so I'll add it there for now.